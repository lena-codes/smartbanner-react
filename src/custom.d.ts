import { FunctionComponent, SVGAttributes } from 'react'

/**
 * Default CSS definition for typescript,
 * will be overridden with file-specific definitions by rollup
 */
declare module '*.css' {
  const content: { [className: string]: string }
  export default content
}

interface SvgrComponent extends FunctionComponent<SVGAttributes<SVGElement>> { }

declare module '*.svg' {
  const svgUrl: string
  const svgComponent: SvgrComponent
  export default svgUrl
  export { svgComponent as ReactComponent }
}
