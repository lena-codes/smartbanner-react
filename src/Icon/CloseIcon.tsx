import React, { SVGProps } from 'react'

export const CloseIcon: React.FC<SVGProps<SVGSVGElement>> = ({
  fill,
  height,
  width,
  ...rest
}) => {
  return (
    <svg
      fill={fill ?? '#000000'}
      xmlns='http://www.w3.org/2000/svg'
      viewBox={'0 0 50 50'}
      width={width ?? '16px'}
      height={height ?? '16px'}
      {...rest}
    >
      <path d='M 7.71875 6.28125 L 6.28125 7.71875 L 23.5625 25 L 6.28125 42.28125 L 7.71875 43.71875 L 25 26.4375 L 42.28125 43.71875 L 43.71875 42.28125 L 26.4375 25 L 43.71875 7.71875 L 42.28125 6.28125 L 25 23.5625 Z' />
    </svg>
  )
}
