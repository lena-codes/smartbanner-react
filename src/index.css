.smartbanner-link {
  margin-right: 6px;
  font-size: large;
  text-decoration: none;
}

.smartbanner-link-light {
  color: #0000ee;
}

.smartbanner-link-dark {
  color: whitesmoke;
}

.smartbanner-close-button {
  cursor: pointer;
  background: transparent;
  border: none;
  padding: 6px;
}

.smartbanner-close-button:hover {
  background-color: gray;
  border-radius: 20%;
}

.smartbanner-img {
  height: 64px;
  width: 64px;
  border-radius: 20%;
  padding-left: 4px;
}

.smartbanner {
  display: flex;
  flex-direction: row;
  min-width: 0;
  max-width: 100vw;
  min-height: 65px;
  justify-content: space-between;
  align-items: center;
  padding: 6px 14px 6px 14px;
}

.smartbanner-row {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
}

.smartbanner-column {
  display: flex;
  flex-direction: column;
  justify-content: center;
  line-height: normal;
}

.smartbanner-light {
  color: grey;
  background: whitesmoke;
}

.smartbanner-dark {
  background: rgb(31, 31, 31);
  color: rgb(191, 191, 191);
}

.smartbanner-title {
  font-size: large;
  margin-top: 0;
  margin-bottom: 3px;
}

.smartbanner-title-light {
  color: black;
}

.smartbanner-title-dark {
  color: whitesmoke;
}

.smartbanner-text {
  margin-top: 0;
  margin-bottom: 0;
}

.smartbanner-space-right {
  margin-right: 10px;
}

.smartbanner-space-left {
  margin-left: 10px;
}

.smartbanner-small-font {
  font-size: small;
}

.smartbanner-medium-font {
  font-size: medium;
}