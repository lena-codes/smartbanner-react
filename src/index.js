import React from 'react'
import CloseIcon from '../src/Icon/CloseIcon.js'
import './index.css'

const SmartBanner = ({
  src,
  name,
  author,
  description,
  buttonText,
  buttonLink,
  background,
  onClose,
  textColor
}) => {
  return (
    <div
      className='banner'
      style={{
        background: background ?? 'WhiteSmoke'
      }}
    >
      <div className='row'>
        <button type='button' onClick={onClose}>
          <CloseIcon />
        </button>
        <div
          className='row'
          style={{
            marginRight: '10px'
          }}
        >
          <img src={src} role='presentation' />
          <div
            style={{
              display: 'flex',
              flexDirection: 'column',
              justifyContent: 'center',
              marginLeft: '6px'
            }}
          >
            <p
              style={{
                fontSize: '16px',
                marginTop: '0',
                marginBottom: '5px',
                height: '15px'
              }}
            >
              {name}
            </p>
            <p
              style={{
                marginTop: '0',
                marginBottom: '0',
                height: '15px',
                fontSize: '12px',
                color: textColor ?? 'grey'
              }}
            >
              {author}
            </p>
            <p
              style={{
                marginTop: '0',
                marginBottom: '5px',
                height: '15px',
                fontSize: '14px',
                color: textColor ?? 'grey'
              }}
            >
              {description}
            </p>
          </div>
        </div>
      </div>
      <a href={buttonLink} target='_blank' role='button'>
        {buttonText}
      </a>
    </div>
  )
}

export default SmartBanner
