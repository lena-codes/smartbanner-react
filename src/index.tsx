import React from 'react'
import { CloseIcon } from './Icon/CloseIcon'
import './index.css'

interface SmartBannerProps {
  src: string
  title: string
  author: string
  description: string
  buttonText: string
  buttonLink: string
  theme?: Theme
  background?: string
  onClose?: () => void
  textColor?: string
  titleColor?: string
}

export type Theme = 'light' | 'dark'

export const SmartBanner: React.FC<SmartBannerProps> = ({
  src,
  title: name,
  author,
  description,
  buttonText,
  buttonLink,
  theme = 'light',
  onClose
}) => {
  return (
    <div className={`smartbanner smartbanner-${theme}`}>
      <div className='smartbanner-row'>
        {onClose && (
          <button
            type='button'
            onClick={onClose}
            className='smartbanner-close-button'
          >
            <CloseIcon theme={theme} />
          </button>
        )}
        <div className='smartbanner-row smartbanner-space-right'>
          <img src={src} role='presentation' className='smartbanner-img' />
          <div className='smartbanner-column smartbanner-space-left'>
            <p className={`smartbanner-title smartbanner-title-${theme}`}>
              {name}
            </p>
            <p className='smartbanner-text smartbanner-small-font'>{author}</p>
            <p className='smartbanner-text smartbanner-medium-font'>
              {description}
            </p>
          </div>
        </div>
      </div>
      <a
        href={buttonLink}
        className={`smartbanner-link-${theme} smartbanner-link`}
        target='_blank'
        role='button'
      >
        {buttonText}
      </a>
    </div>
  )
}
