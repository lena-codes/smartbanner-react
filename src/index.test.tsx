import React from 'react'
import renderer, { ReactTestInstance } from 'react-test-renderer'

import { SmartBanner } from '../src/index'
import { CloseIcon } from './Icon/CloseIcon'

describe('Snapshot Tests', () => {
  it('renders correctly for all required parameters', () => {
    const tree = renderer
      .create(
        <SmartBanner
          title='test'
          author='testAuthor'
          buttonLink='google.com'
          buttonText='Go'
          description='this is a test'
          src='random-image.png'
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders light theme as default', () => {
    const tree = renderer
      .create(
        <SmartBanner
          title='test'
          author='testAuthor'
          buttonLink='google.com'
          buttonText='Go'
          description='this is a test'
          src='random-image.png'
          onClose={() => {}}
        />
      )
      .toJSON()
    const tree2 = renderer
      .create(
        <SmartBanner
          title='test'
          author='testAuthor'
          buttonLink='google.com'
          buttonText='Go'
          description='this is a test'
          src='random-image.png'
          theme='light'
          onClose={() => {}}
        />
      )
      .toJSON()
    expect(tree!.toString()).toEqual(tree!.toString())
  })
  it('renders correctly for dark theme', () => {
    const tree = renderer
      .create(
        <SmartBanner
          title='test'
          author='testAuthor'
          buttonLink='google.com'
          buttonText='Go'
          description='this is a test'
          src='random-image.png'
          theme='dark'
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when all theme overwrite parameters are set', () => {
    const overwriteBackgroundColor = 'green'
    const overwriteButtonColor = 'red'
    const overwriteCloseIconColor = 'yellow'
    const overwriteTextColor = 'pink'
    const overwriteTitleColor = 'purple'

    const testRenderer = renderer.create(
      <SmartBanner
        title='test'
        author='testAuthor'
        buttonLink='google.com'
        buttonText='Go'
        description='this is a test'
        src='random-image.png'
        background={overwriteBackgroundColor}
        buttonColor={overwriteButtonColor}
        closeIconColor={overwriteCloseIconColor}
        onClose={() => alert('button clicked!')}
        textColor={overwriteTextColor}
        titleColor={overwriteTitleColor}
        theme='light'
      />
    )
    const testInstance = testRenderer.root
    const tree = testRenderer.toJSON()

    expect(tree).toMatchSnapshot()
    expect(testInstance.findByType('svg').props.fill).toEqual(
      overwriteCloseIconColor
    )
    expect(testInstance.findByType('a').props.style.color).toEqual(
      overwriteButtonColor
    )

    testInstance
      .findAllByProps({ className: 'smartbanner-text' })
      .forEach((inst) =>
        expect(inst.props.style.color).toEqual(overwriteTextColor)
      )
    expect(
      testInstance.findByProps({
        className: 'smartbanner-title smartbanner-title-light'
      }).props.style.color
    ).toEqual(overwriteTitleColor)
  })
})
