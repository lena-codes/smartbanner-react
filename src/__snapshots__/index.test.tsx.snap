// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Tests renders correctly for all required parameters 1`] = `
<div
  className="smartbanner smartbanner-light"
  style={Object {}}
>
  <div
    className="smartbanner-row"
  >
    <div
      className="smartbanner-row smartbanner-space-right"
    >
      <img
        className="smartbanner-img"
        role="presentation"
        src="random-image.png"
      />
      <div
        className="smartbanner-column smartbanner-space-left"
      >
        <p
          className="smartbanner-title smartbanner-title-light"
          style={Object {}}
        >
          test
        </p>
        <p
          className="smartbanner-text smartbanner-small-font"
          style={Object {}}
        >
          testAuthor
        </p>
        <p
          className="smartbanner-text smartbanner-medium-font"
          style={Object {}}
        >
          this is a test
        </p>
      </div>
    </div>
  </div>
  <a
    className="smartbanner-link-light smartbanner-link"
    href="google.com"
    rel="noreferrer"
    role="button"
    style={Object {}}
    target="_blank"
  >
    Go
  </a>
</div>
`;

exports[`Snapshot Tests renders correctly for dark theme 1`] = `
<div
  className="smartbanner smartbanner-dark"
  style={Object {}}
>
  <div
    className="smartbanner-row"
  >
    <div
      className="smartbanner-row smartbanner-space-right"
    >
      <img
        className="smartbanner-img"
        role="presentation"
        src="random-image.png"
      />
      <div
        className="smartbanner-column smartbanner-space-left"
      >
        <p
          className="smartbanner-title smartbanner-title-dark"
          style={Object {}}
        >
          test
        </p>
        <p
          className="smartbanner-text smartbanner-small-font"
          style={Object {}}
        >
          testAuthor
        </p>
        <p
          className="smartbanner-text smartbanner-medium-font"
          style={Object {}}
        >
          this is a test
        </p>
      </div>
    </div>
  </div>
  <a
    className="smartbanner-link-dark smartbanner-link"
    href="google.com"
    rel="noreferrer"
    role="button"
    style={Object {}}
    target="_blank"
  >
    Go
  </a>
</div>
`;

exports[`Snapshot Tests renders correctly when all theme overwrite parameters are set 1`] = `
<div
  className="smartbanner smartbanner-light"
  style={
    Object {
      "background": "green",
    }
  }
>
  <div
    className="smartbanner-row"
  >
    <button
      className="smartbanner-close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="yellow"
        height="16px"
        viewBox="0 0 50 50"
        width="16px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M 7.71875 6.28125 L 6.28125 7.71875 L 23.5625 25 L 6.28125 42.28125 L 7.71875 43.71875 L 25 26.4375 L 42.28125 43.71875 L 43.71875 42.28125 L 26.4375 25 L 43.71875 7.71875 L 42.28125 6.28125 L 25 23.5625 Z"
        />
      </svg>
    </button>
    <div
      className="smartbanner-row smartbanner-space-right"
    >
      <img
        className="smartbanner-img"
        role="presentation"
        src="random-image.png"
      />
      <div
        className="smartbanner-column smartbanner-space-left"
      >
        <p
          className="smartbanner-title smartbanner-title-light"
          style={
            Object {
              "color": "purple",
            }
          }
        >
          test
        </p>
        <p
          className="smartbanner-text smartbanner-small-font"
          style={
            Object {
              "color": "pink",
            }
          }
        >
          testAuthor
        </p>
        <p
          className="smartbanner-text smartbanner-medium-font"
          style={
            Object {
              "color": "pink",
            }
          }
        >
          this is a test
        </p>
      </div>
    </div>
  </div>
  <a
    className="smartbanner-link-light smartbanner-link"
    href="google.com"
    rel="noreferrer"
    role="button"
    style={
      Object {
        "color": "red",
      }
    }
    target="_blank"
  >
    Go
  </a>
</div>
`;
